app/js/enhancedDate.js

```diff
@@ -1,5 +1,76 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function(window, undefined) {
 
+	var _date;
+	var weekday = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+	var month = ["January", "February", "March", "April", "May", "June", "July", "Auguest", "September", "October", "Novenber", "December"];
+	var current = new Date();
+	var setDate = false;
+
+	function _setDate(oDate) {
+		if(oDate instanceof Date) {
+			_date = oDate;
```

Comment:
Because objects are pass-by-reference, assigning oDate to _date means _date could be modified by modifying the object that is passed in. Better to simply instantiate a new Date object, passing in oDate to the Date constructor.